src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -157,36 +158,51 @@ private String verifyConnectionInternal(
           + clientVersionString;
     }
     for (final String s : getBadWords()) {
-      if (clientName.toLowerCase().contains(s.toLowerCase())) {
+      if (user.getUsername().toLowerCase().contains(s.toLowerCase())) {
         return ErrorMessages.THATS_NOT_A_NICE_NAME;
       }
     }
-    if (hashedMac == null) {
-      return ErrorMessages.UNABLE_TO_OBTAIN_MAC;
-    }
```

Comment:
I removed this guard because it appears to be an impossible condition.  `ClientMessenger` enforces that the hashed MAC is non-`null` before sending.

From a security perspective, we could have a malicious client send a `null` hashed MAC.  As things stand now, an NPE would be thrown on the server earlier in the call stack.  I don't know at this time if that would bring down the entire lobby or just abort the servicing thread (I'm assuming the latter).  Obviously, that could lead to a DoS vulnerability.  IMO, the correct defense against such an attack should be in `ServerQuarantineConversation` where the network protocol is managed.